src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2352,6 +2352,11 @@ private BoundExpression BindCastCore(ExpressionSyntax node, BoundExpression oper
                         GenerateImplicitConversionError(diagnostics, operand.Syntax, conversion, operand, targetType);
                         return;
                     }
+                case BoundKind.UnconvertedAddressOfOperator:
+                    {
+                        diagnostics.Add(ErrorCode.ERR_AddressOfToNonFunctionPointer , syntax.Location, ((BoundUnconvertedAddressOfOperator)operand).Operand.Name, targetType);
```

Comment:
>ERR_AddressOfToNonFunctionPointer [](start = 50, length = 33)

How do we know the target type is not a function pointer or delegate? Consider adding explicit check.